### PR TITLE
internal/scorecard/alpha: refactor for more intuitive usage

### PR DESCRIPTION
**Pull request by joelanford**
* `TestBundle` type is removed. Functions that used it now use a
  `registry.Bundle` directly

* Bundle helpers moved to `scorecardutil` package and refactored
  to parse the new bundle directory format

* Helper for extracting ALM examples added to `scorecardutil`

* Minor test tweaks and fixes.

### CODE REVIEW

File: internal/scorecard/alpha/config_test.go [hunk 1]
```diff
@@ -30,11 +30,12 @@ func TestInvalidConfigPath(t *testing.T) {
...
-				return
-			} else if err != nil && !c.wantError {
-				t.Errorf("Wanted result but got error: %v", err)
+			if err == nil && c.wantError {
+				t.Fatalf("Wanted error but got no error")
+			} else if err != nil {
+				if !c.wantError {
+					t.Fatalf("Wanted result but got error: %v", err)
+				}
 				return
```
Comment (camilamacedo86) [inline internal/scorecard/alpha/config_test.go:39]:
```suggestion
		if (err != nil) != c.wantError {
				t.Errorf("TestInvalidConfigPath error = %v, wantErr %v", err, tt.wantErr)
				return
			}

```

File: internal/scorecard/alpha/labels_test.go [hunk 2]
```diff
@@ -46,11 +46,12 @@ func TestEmptySelector(t *testing.T) {
...
-				return
-			} else if err != nil && !c.wantError {
-				t.Errorf("Wanted result but got error: %v", err)
+			if err == nil && c.wantError {
+				t.Fatalf("Wanted error but got no error")
+			} else if err != nil {
+				if !c.wantError {
+					t.Fatalf("Wanted result but got error: %v", err)
+				}
 				return
```
Comment (camilamacedo86) [inline internal/scorecard/alpha/labels_test.go:55]: same.

File: internal/scorecard/alpha/scorecardutil/bundle.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+//	dependenciesData, err := ioutil.ReadFile(path)
+//	if err != nil {
+//		return nil, err
+//	}
+//	dependencies := registry.DependenciesFile{}
+//	if err := yaml.Unmarshal(dependenciesData, &dependencies); err != nil {
+//		return nil, err
+//	}
+//	return &dependencies, nil
+//}
```
Comment (camilamacedo86) [inline internal/scorecard/alpha/scorecardutil/bundle.go:88]: If is commented could not be removed?

File: internal/scorecard/alpha/scorecardutil/bundle.go [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+)
+
+// TODO(joelanford): This function should probably be upstreamed into
+//     operator-framework/operator-registry and improved. Much of the code
+//     upstream already exists, but not in a consumable way. See:
+//     https://github.com/operator-framework/operator-registry/blob/2bdcd07e6f9ea23a35f1fdbb6a7c6584ec53a0a1/pkg/registry/imageinput.go#L25-L81
+//
+// LoadBundleDirectory parses a Bundle from a given on-disk path.
+func LoadBundleDirectory(bundlePath string) (*registry.Bundle, error) {
+	metadataDir := filepath.Join(bundlePath, "metadata")
```
Comment (estroz) [inline internal/scorecard/alpha/scorecardutil/bundle.go:35]: Nit: use bundle package constants, ex. [`bundle.MetadataDir`](https://github.com/operator-framework/operator-registry/blob/master/pkg/lib/bundle/generate.go#L25).

File: internal/scorecard/alpha/scorecardutil/bundle.go [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+	annotations := registry.AnnotationsFile{}
+	if err := yaml.Unmarshal(annotationsData, &annotations); err != nil {
+		return nil, err
+	}
+	return &annotations, nil
+}
+
+// TODO(joelanford): bump to Kubernetes v1.18 and latest operator-registry
+//  to get access to dependencies file definitions.
+//func decodeDependenciesFile(path string) (*registry.DependenciesFile, error) {
```
Comment (estroz) [inline internal/scorecard/alpha/scorecardutil/bundle.go:78]: This is a relatively new item available in `metadata/`. Do does the scorecard need to "understand" dependencies just yet?

**Pull request closed without merging**